### ISSUE TITLE
backend tfjs-backend-webgpu is missing several common kernel ops

### ISSUE THREAD

**Issue by vladmandic** [tensorflow/tfjs#5496]
after converting my existing app to run with `tfjs-backend-webgpu` built from main branch  
(latest released version `tfjs-backend-webgpu 0.0.1-alpha.7` is just too old),  
i've run into issues running some of the models (all of them work perfectly using `webgl` backend):

> Kernel 'TopK' not registered for backend 'webgpu'
> Kernel 'SplitV' not registered for backend 'webgpu'
> Kernel 'FlipLeftRight' not registered for backend 'webgpu'
> Kernel 'RotateWithOffset' not registered for backend 'webgpu'

note: with remaining running models, with `webgpu` inference is 300-400ms while with `webgl` it's 400-550ms  
which makes `webgpu` ~30% faster than `webgl`!

btw, i'll try using webgpu in web workers next as single thread does not come even close of saturating gpu  
(average utilization is ~25%)

**Comment by qjia7** [comment]
@xhcao Please take a look, thanks.

**Comment by carrycooldude** [comment]
I can work on this if you say...

**Comment by gyagp** [comment]
@carrycooldude, any contribution to WebGPU backend is highly appreciated! Once you have something ready for review, you may @qjia7 and we can invite other related reviewers. Thanks!

**Comment by xhcao** [comment]
Hi, @carrycooldude Are you implementing these kernel ops? You could submit several patches, and each patch only implements one kernel op.
I know It is difficult to implement these ops, if you have some problems, we could discuss the solutions.

**Comment by carrycooldude** [comment]
Sure @xhcao I started working on it, I will let you know if I having any issue and Also If possible can you make a project tab for me for the missing kernels

**Comment by xhcao** [comment]
Hi, @carrycooldude, How many operators had been implemented? Do you need help from me to implement them? I think @vladmandic is waiting us to implement these operator, and verify it work on his refactoring. https://github.com/tensorflow/tfjs/issues/5468
You could refer the implementations from webgl backend to implement the first version and optimize them in feature.

**Comment by carrycooldude** [comment]
Can you give me some PR for reference

**Comment by xhcao** [comment]
@carrycooldude Below links implement these kernels on webgl backend, you could refer them.
https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-webgl/src/kernels/TopK.ts
https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-webgl/src/kernels/SplitV.ts
https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-webgl/src/kernels/FlipLeftRight.ts
https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-webgl/src/kernels/RotateWithOffset.ts

**Comment by carrycooldude** [comment]
Thanks I will start working on this

**Comment by xhcao** [comment]
Hi, @carrycooldude, how about the process of these kernels? Currently, I have spare time, and could help you to implement the kernels, which have been not started to implement.

**Comment by carrycooldude** [comment]
Yes , Actually Want to understand the webgl code implementation with webgpu in order to get which Files I have to import to implement those kernels

**Comment by xhcao** [comment]
Could I take over these two kernels ('FlipLeftRight' and  'RotateWithOffset')? We could implement them synchronously.

**Comment by carrycooldude** [comment]
Sure , Also of possible can you guide me through your implementation it will be great too 😇

**Comment by qjia7** [comment]
@carrycooldude You can refer to this one #5585, which implements a new kernel `DepthToSpace` for webgpu.

**Comment by carrycooldude** [comment]
Thanks @qjia7

**Comment by xhcao** [comment]
@carrycooldude , I had implemented 'FlipLeftRight' and 'RotateWithOffset' kernels, https://github.com/tensorflow/tfjs/pull/5649 ,you could refer to them.

**Comment by xhcao** [comment]
Thanks yunfei for implementing the `split` kernel. Now, only `topK` kernel is unimplemented, but it is the most difficult one, the algorithm of webgl backend is based on the paper https://anilshanbhag.in/static/papers/gputopk_sigmod18.pdf, I will  investigate and implement this operator with @carrycooldude, and make some optimizations in future according to webgpu features.
